### PR TITLE
Add help content for services

### PR DESCRIPTION
**Pull request by lucasmoura**
Currently, help command for the service does not provide only a basic information for the services. We are now adding the right content for all those services when we run the help command.

### CODE REVIEW

File: help_data.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    help: |
+      esm-apps is enabled by default on entitled workloads. It provides access to a
+      private ppa which includes available high and critical CVE fixes for
+      Ubuntu LTS packages in the Ubuntu Main and Ubuntu Universe repositories from
+      the Ubuntu LTS release date until its end of life. You can find out more about
+      the esm service at https://ubuntu.com/security/esm
+
+livepatch:
+    help: |
+      livepatch provides selected high and critical kernel cve fixes and other
```
Comment (blackboxsw) [inline help_data.yaml:19]:
```suggestion
      livepatch provides selected high and critical kernel CVE fixes and other
```

File: uaclient/entitlements/base.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
     ContractStatus,
     UserFacingStatus,
 )
-from uaclient.defaults import DEFAULT_HELP_FILE, DEFAULT_HELP
```
Comment (blackboxsw) [inline uaclient/entitlements/base.py:25]: You should also delete the definition uaclient.defaults.DEFAULT_HELP

File: help_data.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+     esm-infra is enabled by default on entitled workloads. It provides access to a
+     private ppa which includes available high and critical CVE fixes for Ubuntu LTS
+     packages in the Ubuntu Main repository between the end of the standard (free)
+     Ubuntu LTS security maintenance and its end of life. You can find our more
+     about the esm service at https://ubuntu.com/security/esm.
+
+esm-apps:
+    help: |
+      esm-apps is enabled by default on entitled workloads. It provides access to a
+      private ppa which includes available high and critical CVE fixes for
```
Comment (blackboxsw) [inline help_data.yaml:12]:
```suggestion
      private PPA which includes available high and critical CVE fixes for
```

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Looks pretty good for a first pass on help:
*  I think we will also need to update UAEntitlement.help_doc_url for all entitlements.
* Also given that this is help text for sevices, shouldn't the service itself be the full capitalized name? Like Livepatch instead of livepatch, FIPS instead of fips, and "UA Apps: Extended Security Maintenance" instead of esm-apps?
* Note I have a fix in the help logic to avoid trying to render entitlements which are not defined in ENTITLEMENT_CLASS_BY_NAME


```suggestion
diff --git a/help_data.yaml b/help_data.yaml
index 6ff8d06..3b80aa3 100644
--- a/help_data.yaml
+++ b/help_data.yaml
@@ -1,7 +1,7 @@
 esm-infra:
    help: |
      esm-infra is enabled by default on entitled workloads. It provides access to a
-     private ppa which includes available high and critical CVE fixes for Ubuntu LTS
+     private PPA which includes available high and critical CVE fixes for Ubuntu LTS
      packages in the Ubuntu Main repository between the end of the standard (free)
      Ubuntu LTS security maintenance and its end of life. You can find our more
      about the esm service at https://ubuntu.com/security/esm.
@@ -9,14 +9,14 @@ esm-infra:
 esm-apps:
     help: |
       esm-apps is enabled by default on entitled workloads. It provides access to a
-      private ppa which includes available high and critical CVE fixes for
+      private PPA which includes available high and critical CVE fixes for
       Ubuntu LTS packages in the Ubuntu Main and Ubuntu Universe repositories from
       the Ubuntu LTS release date until its end of life. You can find out more about
       the esm service at https://ubuntu.com/security/esm
 
 livepatch:
     help: |
-      livepatch provides selected high and critical kernel cve fixes and other
+      Livepatch provides selected high and critical kernel CVE fixes and other
       non-security bug fixes as kernel livepatches. Livepatches are applied
       without rebooting a machine which drastically limits the need for
       unscheduled system reboots. Due to the nature of fips compliance,
diff --git a/uaclient/config.py b/uaclient/config.py
index bcdcd40..30f0317 100644
--- a/uaclient/config.py
+++ b/uaclient/config.py
@@ -451,7 +451,7 @@ class UAConfig:
         response_dict["name"] = name
 
         for resource in resources:
-            if resource["name"] == name:
+            if resource["name"] == name and name in ENTITLEMENT_CLASS_BY_NAME:
                 help_resource = resource
                 help_ent_cls = ENTITLEMENT_CLASS_BY_NAME.get(name)
                 help_ent = help_ent_cls(self)
diff --git a/uaclient/defaults.py b/uaclient/defaults.py
index 01b63fc..129bc14 100644
--- a/uaclient/defaults.py
+++ b/uaclient/defaults.py
@@ -17,7 +17,3 @@ CONFIG_DEFAULTS = {
     "log_level": "INFO",
     "log_file": "/var/log/ubuntu-advantage.log",
 }
-
-DEFAULT_HELP = {
-    "esm-infra": {"help": "esm-infra help\nInformation about esm-infra"}
-}
csmith@uptown:~/src/ubuntu-advantage-client$ git diff 
diff --git a/help_data.yaml b/help_data.yaml
index 6ff8d06..3b80aa3 100644
--- a/help_data.yaml
+++ b/help_data.yaml
@@ -1,7 +1,7 @@
 esm-infra:
    help: |
      esm-infra is enabled by default on entitled workloads. It provides access to a
-     private ppa which includes available high and critical CVE fixes for Ubuntu LTS
+     private PPA which includes available high and critical CVE fixes for Ubuntu LTS
      packages in the Ubuntu Main repository between the end of the standard (free)
      Ubuntu LTS security maintenance and its end of life. You can find our more
      about the esm service at https://ubuntu.com/security/esm.
@@ -9,14 +9,14 @@ esm-infra:
 esm-apps:
     help: |
       esm-apps is enabled by default on entitled workloads. It provides access to a
-      private ppa which includes available high and critical CVE fixes for
+      private PPA which includes available high and critical CVE fixes for
       Ubuntu LTS packages in the Ubuntu Main and Ubuntu Universe repositories from
       the Ubuntu LTS release date until its end of life. You can find out more about
       the esm service at https://ubuntu.com/security/esm
 
 livepatch:
     help: |
-      livepatch provides selected high and critical kernel cve fixes and other
+      Livepatch provides selected high and critical kernel CVE fixes and other
       non-security bug fixes as kernel livepatches. Livepatches are applied
       without rebooting a machine which drastically limits the need for
       unscheduled system reboots. Due to the nature of fips compliance,
diff --git a/uaclient/config.py b/uaclient/config.py
index bcdcd40..30f0317 100644
--- a/uaclient/config.py
+++ b/uaclient/config.py
@@ -451,7 +451,7 @@ class UAConfig:
         response_dict["name"] = name
 
         for resource in resources:
-            if resource["name"] == name:
+            if resource["name"] == name and name in ENTITLEMENT_CLASS_BY_NAME:
                 help_resource = resource
                 help_ent_cls = ENTITLEMENT_CLASS_BY_NAME.get(name)
                 help_ent = help_ent_cls(self)
diff --git a/uaclient/defaults.py b/uaclient/defaults.py
index 01b63fc..129bc14 100644
--- a/uaclient/defaults.py
+++ b/uaclient/defaults.py
@@ -17,7 +17,3 @@ CONFIG_DEFAULTS = {
     "log_level": "INFO",
     "log_file": "/var/log/ubuntu-advantage.log",
 }
-
-DEFAULT_HELP = {
-    "esm-infra": {"help": "esm-infra help\nInformation about esm-infra"}
-}
```

**Comment by lucasmoura** [comment]
@blackboxsw I have updated the code. But there still some items that I think should be discussed:

1) I have asked about the output of the command and it seems that we should only follow the 80 characters at most in a line. However, the command output was suggested to be changed to:

```
Name:
esm-infra

Available:
yes

Help:
UA Infra: Extended Security Maintenance is enabled by default on entitled
workloads. It provides access to a private PPA which includes available
high and critical CVE fixes for Ubuntu LTS packages in the Ubuntu Main
repository between the end of the standard (free) Ubuntu LTS security
maintenance and its end of life. You can find our more about the esm
service at https://ubuntu.com/security/esm.
```

If that is not okay, I can revert back to the old format.

2) Should we capitalize the `fips-updates` command too ?  Like `FIPS-updates`

3) It seems that the `cis-audit` resource name is not `cis-audit`, but only `cis`. In that situation, should we inform the contract team about this issue or just adapt the code on our side ?

File: features/attached_commands.feature [hunk 4]
```diff
@@ -293,30 +293,44 @@ Feature: Command behaviour when attached to an UA subscription
...
             """
-            {"name": "esm-infra", "entitled": "yes", "status": "enabled", "help": "esm-infra help\nInformation about esm-infra"}
+            {"name": "esm-infra", "entitled": "yes", "status": "enabled", "help": "UA Infra: Extended Security Maintenance is enabled by default on entitled\nworkloads. It provides access to a private PPA which includes available\nhigh and critical CVE fixes for Ubuntu LTS packages in the Ubuntu Main\nrepository between the end of the standard (free) Ubuntu LTS security\nmaintenance and its end of life. You can find out more about the esm\nservice at https://ubuntu.com/security/esm.\n"}
             """
         When I run `ua help invalid-service` with sudo
         Then I will see the following on stdout:
             """
-            name: invalid-service
-            help: No help available for service "invalid-service"
+            Name:
```
Comment (blackboxsw) [inline features/attached_commands.feature:329]: Is this really desired output versus a usage error?


```
usage: ua help [service_name] [flags]
Unknown service_name 'a'.
```

File: help_data.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      fips-updates installs FIPS updates modules including all security patches
+      for those modules that have been provided since their certification date.
+      You can find out more at ubuntu.com/security/fips
+
+livepatch:
+    help: |
+      Livepatch provides selected high and critical kernel CVE fixes and other
+      non-security bug fixes as kernel livepatches. Livepatches are applied
+      without rebooting a machine which drastically limits the need for
+      unscheduled system reboots. Due to the nature of fips compliance,
```
Comment (blackboxsw) [inline help_data.yaml:62]:
```suggestion
      unscheduled system reboots. Due to the nature of FIPS compliance,
```

File: help_data.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      livepatches cannot be enabled on fips-enabled systems. You can find out
```
Comment (blackboxsw) [inline help_data.yaml:63]:
```suggestion
      livepatches cannot be enabled on FIPS-enabled systems. You can find out
```

**Review by blackboxsw — CHANGES_REQUESTED** [review]

File: features/attached_commands.feature [hunk 7]
```diff
@@ -293,30 +293,44 @@ Feature: Command behaviour when attached to an UA subscription
         https://esm.ubuntu.com/ubuntu/ trusty-infra-updates/main amd64 Packages
         """
 
+    @wip
```
Comment (blackboxsw) [inline features/attached_commands.feature:296]: Let's drop the wip as well

File: features/attached_commands.feature (hunk 4 above)
Comment (mitechie) [reply features/attached_commands.feature:329]: I would take service out of the wording. Just "no help available for "user-input"

**Comment by lucasmoura** [comment]
@blackboxsw Updated

**Comment by blackboxsw** [comment]
> @blackboxsw I have updated the code. But there still some items that I think should be discussed:
> 
>     1. I have asked about the output of the command and it seems that we should only follow the 80 characters at most in a line. However, the command output was suggested to be changed to:

Newlines for between human readable key:\nvalue pairs makes sense to me. We can (and do) collapse a more machine-readable format in --format json. This should be easy for people to read. If need be we can iterate after testing/review.


>     1. Should we capitalize the `fips-updates` command too ?  Like `FIPS-updates`

FIPS Updates no hyphen
>     2. It seems that the `cis-audit` resource name is not `cis-audit`, but only `cis`. In that situation, should we inform the contract team about this issue or just adapt the code on our side ?

Let's set cis on our side and we when we claim to have CIS-audit support officially we can sort the details if needed.

**Review by blackboxsw — APPROVED** [review]

**Pull request merged**